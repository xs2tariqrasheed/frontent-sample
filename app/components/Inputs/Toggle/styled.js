import styled from 'styled-components';
import theme from '../../../styles/theme';

export const Label = styled.label`
  position: relative;
  display: inline-block;
  font-family: Roboto;
  width: 6rem;
  height: 3rem;
  outline: none;

  input {
    opacity: 0;

    &:checked + div {
      background-color: ${theme.color.primary};
    }

    &:checked + div:before {
      transform: translateX(2.6rem);
    }
  }

  div {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #ccc;
    transition: 0.4s;
    border-radius: 3rem;

    &:before {
      position: absolute;
      content: '';
      height: 2.2rem;
      width: 2.2rem;
      left: 0.6rem;
      bottom: 0.4rem;
      background-color: white;
      transition: 0.4s;
      border-radius: 50%;
    }
  }
`;
