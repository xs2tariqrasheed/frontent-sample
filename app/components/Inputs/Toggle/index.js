import React from 'react';
import { Label } from './styled';

const Toggle = props => (
  <Label>
    <input type="checkbox" {...props} />
    <div />
  </Label>
);

export default Toggle;
