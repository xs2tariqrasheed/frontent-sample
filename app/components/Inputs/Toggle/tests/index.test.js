import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import Toggle from '../index';

describe('<Toggle />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <Toggle />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
