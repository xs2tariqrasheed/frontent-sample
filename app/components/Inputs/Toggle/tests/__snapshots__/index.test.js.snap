// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should render and match the snapshot 1`] = `
<label
  class="styled__Label-sc-2nfokb-0 fXMNSd"
>
  <input
    type="checkbox"
  />
  <div />
</label>
`;
